ui/app/pages/swaps/swaps-gas-customization-modal/swaps-gas-customization-modal.component.js

```diff
@@ -0,0 +1,233 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import PageContainer from '../../../components/ui/page-container'
+import { Tabs, Tab } from '../../../components/ui/tabs'
+import { calcGasTotal } from '../../send/send.utils'
+import {
+  sumHexWEIsToRenderableFiat,
+} from '../../../helpers/utils/conversions.util'
+import AdvancedGasInputs from '../../../components/app/gas-customization/advanced-gas-inputs'
+import BasicTabContent from '../../../components/app/gas-customization/gas-modal-page-container/basic-tab-content'
+
+export default class GasModalPageContainer extends Component {
+  static contextTypes = {
+    t: PropTypes.func,
+    metricsEvent: PropTypes.func,
+    trackEvent: PropTypes.func,
+  }
+
+  static propTypes = {
+    insufficientBalance: PropTypes.bool,
+    gasPriceButtonGroupProps: PropTypes.object,
+    infoRowProps: PropTypes.shape({
+      originalTotalFiat: PropTypes.string,
+      originalTotalEth: PropTypes.string,
+      newTotalFiat: PropTypes.string,
+      newTotalEth: PropTypes.string,
+      sendAmount: PropTypes.string,
+      transactionFee: PropTypes.string,
+      extraInfoRow: PropTypes.shape({ label: PropTypes.string, value: PropTypes.string }),
+    }),
+    onSubmit: PropTypes.func,
+    cancelAndClose: PropTypes.func,
+    customPriceIsSafe: PropTypes.bool,
+    disableSave: PropTypes.bool,
+    customGasLimitMessage: PropTypes.string,
+    customTotalSupplement: PropTypes.string,
+    value: PropTypes.string,
+    conversionRate: PropTypes.string,
+    customGasPrice: PropTypes.string,
+    customGasLimit: PropTypes.string,
+    setSwapsCustomizationModalPrice: PropTypes.func,
+    setSwapsCustomizationModalLimit: PropTypes.func,
+    gasEstimateLoadingHasFailed: PropTypes.bool,
+    averageGasPriceInHexWei: PropTypes.string,
+  }
+
+  state = {
+    gasSpeedType: '',
+  }
+
+  setGasSpeedType (gasPriceInHexWei) {
+    if (gasPriceInHexWei === this.props.averageGasPriceInHexWei) {
```

Comment:
Are we sure that this will always be `true` if the user clicks "Average"? :thinking: 
We could return `gasEstimateType` as a second parameter in `handleGasPriceSelection` to be more certain of this, and to ensure that we don't set `fast` in all cases if we break something.

Reply:
> Are we sure that this will always be true if the user clicks "Average"?

Yes, although it may not be clear in the code. In `swaps-gas-customization-modal.container.js` `averageEstimateData` is included in the `gasButtonInfo` array, and its `priceInHexWei` property is what sets `this.props.averageGasPriceInHexWei`

This same `priceInHexWei` property, by virtue of being part of `gasButtonInfo` is what gets passed to `handleGasPriceSelection` (https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/components/app/gas-customization/gas-price-button-group/gas-price-button-group.component.js#L65)

Reply:
Although, I suppose your "in all cases if we break something" comment juxtaposed with my "although it may not be clear in the code" is what makes the case for your suggested improvement. Because its not clear, someone not aware of what I just explained might break one of the assumptions that makes `gasPriceInHexWei === this.props.averageGasPriceInHexWei` true right now... okay, I'll implement your suggestion

Reply:
Done in fca101cd5